presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java

```diff
@@ -187,19 +199,20 @@ private static void tryDeleteFile(Path path)
     public Optional<Iterator<Page>> get(String serializedPlan, Split split)
     {
         CacheKey key = new CacheKey(serializedPlan, split.getSplitIdentifier());
-        Path path = cache.getIfPresent(key);
-        if (path == null) {
+        CacheEntry cacheEntry = cache.getIfPresent(key);
+        if (cacheEntry == null) {
             fragmentCacheStats.incrementCacheMiss();
             return Optional.empty();
         }
 
         try {
-            InputStream inputStream = newInputStream(path);
-            Iterator<Page> result = readPages(pagesSerde, new InputStreamSliceInput(inputStream));
+            InputStream inputStream = newInputStream(cacheEntry.getPath());
+            Iterator<Page> result = readPages(pagesSerdeFactory.createPagesSerde(), new InputStreamSliceInput(inputStream));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the resource returned by the *InputStreamSliceInput* constructor in a try-finally block. The resource is referenced at line 210. The resource is closed at line 212. There are other execution paths that do not contain closure statements, for example, when FragmentCacheStats.incrementCacheHit throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).